### PR TITLE
Create mergesortedarrays.py

### PR DESCRIPTION
**Pull request by kylekirch**
Merges arrays that are sorted in ascending order. The resulting array contains all elements of both arrays in ascending order.

### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by kylekirch** [comment]
This is my first commit to this repo, the algo I have put together merges arrays that are sorted in ascending order. The resulting array contains all elements of both arrays in ascending order. This algo is commonly used in many different situations.

File: data_structures/arrays/mergesortedarrays.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+def merge_sorted_arrays(arr1: list[int], arr2: list[int]) -> list[int]:
+    """
+    Merge two sorted arrays into a single sorted array.
+    >>> merge_sorted_arrays([1, 3, 5], [2, 4, 6])
+    [1, 2, 3, 4, 5, 6]
+    """
+    merged_array = []
+    i = j = 0
+    while i < len(arr1) and j < len(arr2):
```
Comment (amirsoroush) [inline data_structures/arrays/mergesortedarrays.py:9]: test

**Review by rohan472000 — CHANGES_REQUESTED** [review]
use `black .` for pre-commit failure , also check CONTRIBUTING.md in repo for pre-requisites before making PRs.

File: data_structures/arrays/mergesortedarrays.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
+def merge_sorted_arrays(arr1: list[int], arr2: list[int]) -> list[int]:
+    """
+    Merge two sorted arrays into a single sorted array.
+    >>> merge_sorted_arrays([1, 3, 5], [2, 4, 6])
+    [1, 2, 3, 4, 5, 6]
+    """
```
Comment (rohan472000) [inline data_structures/arrays/mergesortedarrays.py:6]: add more unique doctests with empty lists.

**Review by rohan472000 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**